README.md

```diff
@@ -28,43 +28,80 @@ go mod tidy
+download either a `clouds.yaml` file or an `openrc` bash file that exports all
+of your access details to environment variables. To use the `clouds.yaml` file,
+place it at `~/.config/openstack/clouds.yaml`. To use the `openrc` file, run
+`source openrc` and you will be prompted for your password.
 
 ### Authentication
 
-> NOTE: It is now recommended to use the `clientconfig` package found at
-> https://github.com/gophercloud/utils/tree/master/openstack/clientconfig
-> for all authentication purposes.
->
-> The below documentation is still relevant. clientconfig simply implements
-> the below and presents it in an easier and more flexible way.
-
 Once you have access to your credentials, you can begin plugging them into
-Gophercloud. The next step is authentication, and this is handled by a base
-"Provider" struct. To get one, you can either pass in your credentials
-explicitly, or tell Gophercloud to use environment variables:
+Gophercloud. The next step is authentication, which is handled by a base
+"Provider" struct. There are number of ways to construct such a struct.
+
+**With `gophercloud/utils`**
+
+The [github.com/gophercloud/utils](https://github.com/gophercloud/utils)
+library provides the `clientconfig` package to simplify authentication. It
+provides additional functionality, such as the ability to read `clouds.yaml`
+files. To generate a "Provider" struct using the `clientconfig` package:
+
+```go
+import (
+	"github.com/gophercloud/utils/openstack/clientconfig"
+)
+
+// You can also skip configuring this and instead set 'OS_CLOUD' in your
+// environment
+opts := new(clientconfig.ClientOpts)
+opts.Cloud = "devstack-admin"
+
+provider, err := clientconfig.AuthenticatedClient(opts)
+```
+
+A provider client is a top-level client that all of your OpenStack service
+clients derive from. The provider contains all of the authentication details
+that allow your Go code to access the API - such as the base URL and token ID.
+
+Once we have a base Provider, we inject it as a dependency into each OpenStack
+service. For example, in order to work with the Compute API, we need a Compute
+service client. This can be created like so:
+
+```go
+client, err := clientconfig.NewServiceClient("compute", opts)
+```
+
+**Without `gophercloud/utils`**
+
+> *Note*
+> gophercloud doesn't provide support for `clouds.yaml` file so you need to
+> implement this functionality yourself if you don't wish to use
+> `gophercloud/utils`.
+
+You can also generate a "Provider" struct without using the `clientconfig`
+package from `gophercloud/utils`. To do this, you can either pass in your
+credentials explicitly or tell Gophercloud to use environment variables:
 
 ```go
 import (
-  "github.com/gophercloud/gophercloud"
-  "github.com/gophercloud/gophercloud/openstack"
-  "github.com/gophercloud/gophercloud/openstack/utils"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/utils"
```

Comment:
funny that utils is still a dep in the "without utils" section

Reply:
Whoops, done